src/pages/background/background.js

```diff
@@ -28,6 +28,13 @@ browser.runtime.onInstalled.addListener(async details => {
 
 let tabIdRedirects = {}
 
+// page action
+browser.tabs.onUpdated.addListener(async (id, changeInfo, tabInfo) => {
+	const { pageAction } = await utils.getOptions()
+	if (!pageAction) return;
+	browser.pageAction.show(tabInfo.id);
```

Comment:
```suggestion
	const url = new URL(tabInfo.url)
	await servicesHelper.computeService(url).then(r => {
		if (r) {
			browser.pageAction.show(id)
		}
	})
```
This correctly detects if we are using a service that can be redirected with libredirect. I tested it with reddit, youtube, tenor, and github. There may be some failure points, but it seems to do the trick on the initial test.

Reply:
So, you are inspecting page data or what ?

I am confused between ID and URL right now.

Reply:
There is no need to inspect page content-- this is solely based on the URL of the tab.
`show(id)` tells the browser to show the page action (button in the address bar) for that tab identifier:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/tabs/Tab#id

It looks like `servicesHelper.computeServices` can take that url and figure out if it is a reddit link (or one of the alternative reddit types), or if it is a youtube, tenor, etc/alternative service.
It is based on this bit of code: https://github.com/libredirect/browser_extension/blob/32dc67db284b5803c9925df07f2cc47127acfdf6/src/pages/popup/popup.js#L109C3-L115C4